### ISSUE TITLE
Placement is wrong under wayland

### ISSUE THREAD

**Issue by fmoralesc** [p-e-w/plotinus#5]
It seems like Gtk can't handle the parenting of the window properly.

![captura de pantalla de 2016-11-14 13-23-34](https://cloud.githubusercontent.com/assets/221465/20272745/9941f8ea-aa6d-11e6-943d-b650492e1971.png)

**Comment by jhasse** [comment]
Mutter 3.22.2 fixes "various placement issues on wayland", see https://mail.gnome.org/archives/gnome-shell-list/2016-November/msg00002.html

Can you try if this still happens?

**Comment by fmoralesc** [comment]
I just tried with 3.22.3 and it still happens.

**Comment by trgeiger** [comment]
I can confirm that this also happens under Mutter 3.24.1
![screenshot from 2017-04-15 10-12-38](https://cloud.githubusercontent.com/assets/18582917/25064646/1308b768-21c4-11e7-837b-182c772ca45e.png)

**Comment by p-e-w** [comment]
Since Wayland apparently does not support `CENTER_ON_PARENT` at all (you can see the same issue in many GNOME core apps, e.g. the preference dialog in Gedit), I decided to make the popup window modal.

This matches the behavior of Atom's command palette and solves this and several other Wayland issues.

**Comment by fmoralesc** [comment]
This seems to break placement of plotinus in Xorg, though :/

**Comment by p-e-w** [comment]
@fmoralesc: How? Modal windows work the same across Xorg and Wayland (at least all others do).

Can you provide a screenshot of what you are seeing?

**Comment by fmoralesc** [comment]
I would have supposed so, but this is what I see...
![captura de pantalla de 2017-04-23 17-48-44](https://cloud.githubusercontent.com/assets/221465/25314939/3295ef4e-284d-11e7-8b36-3fd362a9513e.png)

It works perfectly in wayland, though (This is using mutter 3.24.1)

**Comment by p-e-w** [comment]
What desktop is this?

What happens when you open a standard modal dialog (e.g. Nautilus' "About" dialog)?

**Comment by fmoralesc** [comment]
This is Gnome 3.24. 

The 'About' dialog is centered. I also considered whether something in my config was wrong... (There is the /org/gnome/mutter/attach-modal-dialogs key that could affect this, but it doesn't seem to make any difference here...)

![captura de pantalla de 2017-04-23 18-01-38](https://cloud.githubusercontent.com/assets/221465/25315047/eae264aa-284e-11e7-90d5-6d5ffca383ec.png)

**Comment by p-e-w** [comment]
Does adding back the line

```
window_position = Gtk.WindowPosition.CENTER_ON_PARENT;
```

that was removed by the commit fix the issue?

I also notice a font issue in your first screenshot. The arrow symbol that is used to separate menu hierarchy levels appears to have been substituted by a placeholder of sorts. What is your UI font?

Beautiful desktop, by the way! 👍

**Comment by fmoralesc** [comment]
(Thanks!) My UI font is Roboto (the Android font). With DejaVu Sans I get a clearer arrow (in Roboto it is an arrow surrounded by a circle).

With that line, it improves, I get
![captura de pantalla de 2017-04-23 18-19-27](https://cloud.githubusercontent.com/assets/221465/25315192/7caf0314-2851-11e7-9a26-f8c840d1e86f.png)

 It still doesn't seem to behave as a modal window.

**Comment by p-e-w** [comment]
Looks like this needs deeper investigation. I hope this doesn't indicate a limitation for GTK+ modules somehow. The placement is weird, it's vertically centered but not horizontally.

Thank you very much for your help so far!

**Comment by fmoralesc** [comment]
No problem, thanks to you for this software!

**Comment by p-e-w** [comment]
@fmoralesc: The Xorg issue is now fixed!

This was a tough nut to crack. Eventually I noticed that the problem disappeared when I made `PopupWindow` inherit from `Gtk.Dialog` instead of `Gtk.Window` (though that introduced other problems). In https://github.com/GNOME/gtk/blob/master/gtk/ui/gtkdialog.ui I then found the key missing property:

```
type_hint = Gdk.WindowTypeHint.DIALOG;
```

Of course, the whole thing still reeks of a GTK+ bug, but at least it's working now in both Wayland and X11!
